edx_proctoring/api.py

```diff
@@ -81,6 +83,47 @@
 USER_MODEL = get_user_model()
 
 
+def get_onboarding_exam_link(course_id, user, is_learning_mfe):
+    """
+    Get link to the onboarding exam available to the user for given course.
+
+    Parameters:
+    * course_id: id of a course where search for the onboarding exam is performed
+    * user: user for whom the link is built
+    * is_learning_mfe: Boolean, indicates if the url should be built for the learning mfe application.
+
+    Returns: url pointing to the available exam, if there is no onboarding exam available
+    to the user returns empty string.
+    """
+    onboarding_exams = list(ProctoredExam.get_practice_proctored_exams_for_course(course_id).order_by('-created'))
+    if not onboarding_exams or not get_backend_provider(name=onboarding_exams[0].backend).supports_onboarding:
+        onboarding_link = ''
+    else:
+        learning_sequences_service = get_runtime_service('learning_sequences')
```

Comment:
a lot of this is repeated code. Can we move this into a util function somewhere?

Reply:
I think most of it can't be put into separate function, that's what I initially wanted to do but the resulting function just gets messy since in the StudentOnboardingStatusView some of the behaviour is just different

I tried to do what I can in 0ff7163 to minimize repeating of the code, please take a look

Reply:
ahh yes I see. This looks good to me, thanks